### ISSUE TITLE
strict stubbing in MockitoFicture is not used when mocks are declared as test class fields

### ISSUE THREAD

**Issue by Krever** [mockito/mockito-scala#31]
```
class MySpec extends WordSpec with MockitoFixture {
  val myMock = mock[Sth]
  //... testing
}
```

because mocks are created before the session is registered(which happens inside `withFixture`) they are not treated as strict mocks and allow unused stubbing.

**Comment by bbonanno** [comment]
Yes, but that's expected, as you want fresh mocks for each test, this has 2 main reasons

1. It is related on how the original MockitoSession works, it's only aware of the mocks created after it has been started and it drops that state as soon as you close it

2. Each test should start with a fresh, clean, state. If you share mocks across tests you could run into problems by tests finding pre-stubbed behavior made in previous tests, which could lead to false positives, false negatives, or even completelynonsensical behavior that is not always easy to spot, 

What you should do instead to avoid repeating yourself, i.e. having to create the same mocks/stubs over and over in each test is to write it this way.

```scala
class MySpec extends WordSpec with MockitoFixture {
   **trait Setup** {
      val myMock = mock[Sth] 
      myMock.someMethod shouldReturn "something" /*stub common to **all** tests -notice that if it's not used by all of them then the session will find it as an unused stubbing on those-/*
   }

   "some feature" should {
       "test whatever i want" in **new Setup** {
            myMock.someOtherMethod(*) shouldReturn None /*stub specific only to this test/*
             ...test
       }

      "test something else" in **new Setup** {
             myMock.someOtherMethod("expected value") shouldReturn Some("result")  /*stub specific only to this test/*
             ...test
       }
   }
}
```

This give you a fresh new instance of `myMock` for each test but at the same time you only declare the creation/common stubbing once.

**Comment by bbonanno** [comment]
hahaha, nice

I agree, it needs to be properly documented
